### PR TITLE
Adding Description template for log transaction

### PR DESCRIPTION
**Pull request by polipopoliko**

### CODE REVIEW

File: api/email/verify_email_token.go [hunk 1]
```diff
@@ -55,13 +54,24 @@ func VerifyEmailToken(db *sql.DB) http.HandlerFunc {
...
+			Email:  et.Email,
+			Status: "verified",
+		})
+
+		if err != nil {
+			helpers.HTTPError(w, http.StatusBadRequest, "Unable to parse to json")
+			return
+		}
+		fmt.Fprintf(w, string(b))
 	}
```
Comment (yulyulyharuka) [inline api/email/verify_email_token.go]: add return at the end of the http.HandlerFunc

**Review by yulyulyharuka — CHANGES_REQUESTED** [review]

**Review by david1312 — APPROVED** [review]
ok

File: api/email/verify_email_token.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+		err = json.Unmarshal(requestBody, &et)
+		if err != nil {
+			helpers.HTTPError(w, http.StatusBadRequest, err.Error())
+			return
+		}
+
+		if err != nil {
+			helpers.HTTPError(w, http.StatusBadRequest, "Unable to verify email token: "+err.Error())
+			return
+		}
```
Comment (vicimilenia) [inline api/email/verify_email_token.go]: remove this, as the previous err != nil already return the function

File: api/vac/virtual_accounts.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+			Description: models.LogDescriptionVaToMainTemplate(vac.VaBalance, vac.VaNum, vac.AccountNum),
+		})
+		if err != nil {
+			helpers.HTTPError(w, http.StatusBadRequest, "Fail to create log transaction")
+			return
+		}
+	} else {
+		err = database.DeleteVacById(vh.Db, vac.VaId)
+		if err != nil {
+			helpers.HTTPError(w, http.StatusBadRequest, "Fail to revert balance to main account")
```
Comment (vicimilenia) [inline api/vac/virtual_accounts.go]:
```suggestion
			helpers.HTTPError(w, http.StatusBadRequest, "Fail to delete vac(?)")
```

File: database/virtual_accounts.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import (
+	"database/sql"
+
+	"github.com/ndv6/tsaving/models"
+)
+
+func RevertBalanceToAccount(db *sql.DB, va models.VirtualAccounts) (err error) {
+	acc, err := GetAccountByAccountNum(db, va.AccountNum)
+	if err == nil {
+		_, err = db.Exec("UPDATE ACCOUNTS SET account_balance=$1 WHERE account_id=$2;", acc.AccountBalance+va.VaBalance, acc.AccountId)
```
Comment (vicimilenia) [inline database/virtual_accounts.go]:
```suggestion
		_, err = db.Exec("UPDATE accounts SET account_balance=$1 WHERE account_id=$2;", acc.AccountBalance+va.VaBalance, acc.AccountId)
```

File: database/virtual_accounts.go [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+	}
+	return
+}
+
+func DeleteVacById(db *sql.DB, vId int) (err error) {
+	_, err = db.Exec("DELETE FROM VIRTUAL_ACCOUNTS WHERE va_id=$1;", vId)
```
Comment (vicimilenia) [inline database/virtual_accounts.go]:
```suggestion
	_, err = db.Exec("DELETE FROM virtual_accounts WHERE va_id=$1;", vId)
```

File: database/virtual_accounts.go [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+	return
+}
+
+func GetAccountByAccountNum(db *sql.DB, accountNum string) (acc models.Accounts, err error) {
+	err = db.QueryRow("SELECT account_id, account_num, account_balance, created_at FROM ACCOUNTS WHERE account_num=$1", accountNum).Scan(&acc.AccountId, &acc.AccountNum, &acc.AccountBalance, &acc.CreatedAt)
```
Comment (vicimilenia) [inline database/virtual_accounts.go]:
```suggestion
	err = db.QueryRow("SELECT account_id, account_num, account_balance, created_at FROM accounts WHERE account_num=$1", accountNum).Scan(&acc.AccountId, &acc.AccountNum, &acc.AccountBalance, &acc.CreatedAt)
```

File: database/virtual_accounts.go [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+	return
+}
+
+func GetCustomerById(db *sql.DB, id int) (cust models.Customers, err error) {
+	err = db.QueryRow("SELECT cust_id, account_num, email FROM CUSTOMERS WHERE cust_id=$1;", id).Scan(&cust.CustId, &cust.AccountNum, &cust.CustEmail)
```
Comment (vicimilenia) [inline database/virtual_accounts.go]:
```suggestion
	err = db.QueryRow("SELECT cust_id, account_num, email FROM customers WHERE cust_id=$1;", id).Scan(&cust.CustId, &cust.AccountNum, &cust.CustEmail)
```

File: database/virtual_accounts.go [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+	return
+}
+
+func GetVacByAccountNum(db *sql.DB, accountNum string) (va models.VirtualAccounts, err error) {
+	err = db.QueryRow("SELECT va_id, va_num, account_num, va_balance FROM VIRTUAL_ACCOUNTS WHERE account_num=$1", accountNum).Scan(&va.VaId, &va.VaNum, &va.AccountNum, &va.VaBalance)
```
Comment (vicimilenia) [inline database/virtual_accounts.go]:
```suggestion
	err = db.QueryRow("SELECT va_id, va_num, account_num, va_balance FROM virtual_accounts WHERE account_num=$1", accountNum).Scan(&va.VaId, &va.VaNum, &va.AccountNum, &va.VaBalance)
```

**Review by vicimilenia — CHANGES_REQUESTED** [review]

**Review by david1312 — APPROVED** [review]
good to go

File: database/database_virtual_accounts.go [hunk 9]
```diff
@@ -18,8 +19,10 @@ func GetBalanceVA(vaNum string, db *sql.DB) (va models.VirtualAccounts, err erro
 
 func RevertVacBalanceToMainAccount(db *sql.DB, va models.VirtualAccounts) (err error) {
 	acc, err := GetAccountByAccountNum(db, va.AccountNum)
+	fmt.Println(err)
```
Comment (yulyulyharuka) [inline database/database_virtual_accounts.go]: i think you can use log.Println() instead of fmt.Println() to print the error

Comment (yulyulyharuka) [reply database/database_virtual_accounts.go]: or you can delete it i think and throw the error in the handler

File: database/database_virtual_accounts.go [hunk 10]
```diff
@@ -18,8 +19,10 @@ func GetBalanceVA(vaNum string, db *sql.DB) (va models.VirtualAccounts, err erro
...
 	if err == nil {
 		_, err = db.Exec("UPDATE accounts SET account_balance=$1 WHERE account_id=$2;", acc.AccountBalance+va.VaBalance, acc.AccountId)
+		fmt.Println(err)
```
Comment (yulyulyharuka) [inline database/database_virtual_accounts.go]: the same problem here

**Review by yulyulyharuka — APPROVED** [review]
lgtm

**Pull request merged**